python3Packages.mcp: 1.9.4 -> 1.12.2, and fix dependent packages build failures

bryanhonof opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [x] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

bryanhonof committed:
python3Packages.mcp: 1.9.4 -> 1.12.2

I also had to disable some failing tests that I didn't mananage to catch in a
reproducible manner. They tend to happen when the system I'm running
nixpkgs-review on is under some load, but they don't happen all that often when
building the package in isolation.

Co-authored-by: Joshua Peek <josh@users.noreply.github.com>

bryanhonof committed:
python3Packages.openai-agents: 0.0.13 -> 0.2.3

Got rid of nix-update-script, since it didn't seem to be doing anything?

josh commented:
I had opened up #427011, but I haven't had time to review the other dependent failures. @bryanhonof are you interested in joining the maintainer list with me? 😄

bryanhonof committed:
python3Packages.mcp: add bryanhonof as maintainer

bryanhonof committed:
python3Packages.fastmcp: 2.8.0 -> 2.10.6

bryanhonof committed:
python3Packages.fastapi-mcp: 0.3.4 -> 0.3.7

josh commented on pkgs/development/python-modules/mcp/default.nix:
```diff
@@ -129,6 +115,9 @@ buildPythonPackage rec {
     license = lib.licenses.mit;
-    maintainers = with lib.maintainers; [ josh ];
+    maintainers = with lib.maintainers; [
+      bryanhonof
```
🙏🏻

josh commented on pkgs/development/python-modules/fastmcp/default.nix:
```diff
@@ -79,9 +87,11 @@ buildPythonPackage rec {
     "tests/auth/test_oauth_client.py"
     "tests/client/test_openapi.py"
     "tests/client/test_sse.py"
+    "tests/client/test_stdio.py"
     "tests/client/test_streamable_http.py"
     "tests/server/http/test_http_dependencies.py"
     "tests/server/http/test_http_dependencies.py"
+    "tests/utilities/test_mcp_config.py"
```
I'm also seeing these fail on linux too.

```
> FAILED tests/client/test_stdio.py::TestKeepAlive::test_keep_alive_maintains_session_across_multiple_calls - RuntimeError: Client failed to connect: Connection closed
> FAILED tests/client/test_stdio.py::TestKeepAlive::test_keep_alive_false_starts_new_session_across_multiple_calls - RuntimeError: Client failed to connect: Connection closed
> FAILED tests/client/test_stdio.py::TestKeepAlive::test_keep_alive_starts_new_session_if_manually_closed - RuntimeError: Client failed to connect: Connection closed
> FAILED tests/client/test_stdio.py::TestKeepAlive::test_keep_alive_maintains_session_if_reentered - RuntimeError: Client failed to connect: Connection closed
> FAILED tests/client/test_stdio.py::TestKeepAlive::test_close_session_and_try_to_use_client_raises_error - RuntimeError: Client failed to connect: Connection closed
> FAILED tests/utilities/test_mcp_config.py::test_multi_client - assert 0 == 2
> FAILED tests/utilities/test_mcp_config.py::test_multi_client_with_logging - fastmcp.exceptions.ToolError: Unknown tool: test_server_log_test
> FAILED tests/utilities/test_mcp_config.py::test_multi_client_with_elicitation - fastmcp.exceptions.ToolError: Unknown tool: test_server_elicit_test
```

Maybe want to just move them into a common list?

```diff
diff --git a/pkgs/development/python-modules/fastmcp/default.nix b/pkgs/development/python-modules/fastmcp/default.nix
index 17a11432cc..3c274c560a 100644
--- a/pkgs/development/python-modules/fastmcp/default.nix
+++ b/pkgs/development/python-modules/fastmcp/default.nix
@@ -81,18 +81,20 @@
     "test_temporary_settings"
   ];
 
-  disabledTestPaths = lib.optionals stdenv.hostPlatform.isDarwin [
+  disabledTestPaths = [
+    "tests/client/test_stdio.py"
+    "tests/utilities/test_mcp_config.py"
+  ]
+  ++ (lib.optionals stdenv.hostPlatform.isDarwin [
     # RuntimeError: Server failed to start after 10 attempts
     "tests/auth/providers/test_bearer.py"
     "tests/auth/test_oauth_client.py"
     "tests/client/test_openapi.py"
     "tests/client/test_sse.py"
-    "tests/client/test_stdio.py"
     "tests/client/test_streamable_http.py"
     "tests/server/http/test_http_dependencies.py"
     "tests/server/http/test_http_dependencies.py"
-    "tests/utilities/test_mcp_config.py"
-  ];
+  ]);
 
   __darwinAllowLocalNetworking = true;
 
```

bryanhonof commented:
`python3Packages.gradio` just seems to time out on macOS.
It also [doesn't look too healthy in Hydra](https://hydra.nixos.org/eval/1817160?filter=gradio&compare=1817154&full=).

siraben commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 428733`
Commit: `f46dda8630bc55a1556c8cd2557fc1b221446bb5`

---
### `aarch64-darwin`
<details>
  <summary>:x: 47 packages failed to build:</summary>
  <ul>
    <li>litellm</li>
    <li>litellm.dist</li>
    <li>mcp-proxy</li>
    <li>mcp-proxy.dist</li>
    <li>oterm</li>
    <li>oterm.dist</li>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python312Packages.django-mcp-server</li>
    <li>python312Packages.django-mcp-server.dist</li>
    <li>python312Packages.fastapi-mcp</li>
    <li>python312Packages.fastapi-mcp.dist</li>
    <li>python312Packages.fastmcp</li>
    <li>python312Packages.fastmcp.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.mcp</li>
    <li>python312Packages.mcp.dist</li>
    <li>python312Packages.mcpadapt</li>
    <li>python312Packages.mcpadapt.dist</li>
    <li>python312Packages.openai-agents</li>
    <li>python312Packages.openai-agents.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python313Packages.django-mcp-server</li>
    <li>python313Packages.django-mcp-server.dist</li>
    <li>python313Packages.fastapi-mcp</li>
    <li>python313Packages.fastapi-mcp.dist</li>
    <li>python313Packages.fastmcp</li>
    <li>python313Packages.fastmcp.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.mcp</li>
    <li>python313Packages.mcp.dist</li>
    <li>python313Packages.mcpadapt</li>
    <li>python313Packages.mcpadapt.dist</li>
    <li>python313Packages.openai-agents</li>
    <li>python313Packages.openai-agents.dist</li>
    <li>python313Packages.private-gpt</li>
    <li>python313Packages.private-gpt.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>vectorcode</li>
    <li>vectorcode.dist</li>
    <li>vimPlugins.vectorcode-nvim</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python312Packages.mcp</summary>
<pre>ERROR tests/shared/test_streamable_http.py::test_streamablehttp_client_json_response - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_streamablehttp_client_session_termination - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_streamablehttp_client_session_termination_204 - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_streamablehttp_client_resumption - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_streamablehttp_request_context_isolation - RuntimeError: Context-aware server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_streamablehttp_request_context_propagation - RuntimeError: Context-aware server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_streamablehttp_server_sampling - RuntimeError: Server failed to start after 20 attempts
ERROR tests/server/fastmcp/test_integration.py::test_tool_progress[server_transport1] - RuntimeError: Server failed to start after 20 attempts
ERROR tests/server/fastmcp/test_integration.py::test_completion[server_transport1] - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_server_validates_protocol_version_header - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_client_includes_protocol_version_header_after_init - RuntimeError: Context-aware server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_server_backwards_compatibility_no_protocol_version - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_ws.py::test_ws_client_basic_connection - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_client_crash_handled - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_ws.py::test_ws_client_exception_handling - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_ws.py::test_ws_client_happy_request_and_response - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_ws.py::test_ws_client_timeout - RuntimeError: Server failed to start after 20 attempts
ERROR tests/server/fastmcp/test_integration.py::test_fastmcp_quickstart[server_transport0] - RuntimeError: Server failed to start after 20 attempts
ERROR tests/server/fastmcp/test_integration.py::test_fastmcp_quickstart[server_transport1] - RuntimeError: Server failed to start after 20 attempts
== 17 failed, 440 passed, 2 skipped, 1 xfailed, 59 errors in 88.34s (0:01:28) ==</pre>
</details>
<details>
<summary>python313Packages.mcp</summary>
<pre>ERROR tests/shared/test_streamable_http.py::test_streamablehttp_client_resource_read - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_streamablehttp_client_error_handling - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_streamablehttp_client_session_persistence - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_streamablehttp_client_get_stream - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_streamablehttp_client_session_termination - RuntimeError: Server failed to start after 20 attempts
ERROR tests/server/fastmcp/test_integration.py::test_completion[server_transport0] - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_streamablehttp_client_resumption - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_streamablehttp_request_context_propagation - RuntimeError: Context-aware server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_streamablehttp_server_sampling - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_streamablehttp_client_session_termination_204 - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_streamablehttp_request_context_isolation - RuntimeError: Context-aware server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_client_includes_protocol_version_header_after_init - RuntimeError: Context-aware server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_server_validates_protocol_version_header - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_client_crash_handled - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_ws.py::test_ws_client_exception_handling - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_ws.py::test_ws_client_basic_connection - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_streamable_http.py::test_server_backwards_compatibility_no_protocol_version - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_ws.py::test_ws_client_happy_request_and_response - RuntimeError: Server failed to start after 20 attempts
ERROR tests/shared/test_ws.py::test_ws_client_timeout - RuntimeError: Server failed to start after 20 attempts
== 17 failed, 440 passed, 2 skipped, 1 xfailed, 59 errors in 91.52s (0:01:31) ==</pre>
</details>
</details>

siraben commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 428733`
Commit: `f46dda8630bc55a1556c8cd2557fc1b221446bb5`

---
### `x86_64-linux`
<details>
  <summary>:x: 41 packages failed to build:</summary>
  <ul>
    <li>home-assistant-component-tests.mcp</li>
    <li>home-assistant-component-tests.mcp_server</li>
    <li>litellm</li>
    <li>litellm.dist</li>
    <li>mcp-proxy</li>
    <li>mcp-proxy.dist</li>
    <li>oterm</li>
    <li>oterm.dist</li>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
    <li>python313Packages.django-mcp-server</li>
    <li>python313Packages.django-mcp-server.dist</li>
    <li>python313Packages.fastapi-mcp</li>
    <li>python313Packages.fastapi-mcp.dist</li>
    <li>python313Packages.fastmcp</li>
    <li>python313Packages.fastmcp.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.mcp</li>
    <li>python313Packages.mcp.dist</li>
    <li>python313Packages.mcpadapt</li>
    <li>python313Packages.mcpadapt.dist</li>
    <li>python313Packages.openai-agents</li>
    <li>python313Packages.openai-agents.dist</li>
    <li>python313Packages.private-gpt</li>
    <li>python313Packages.private-gpt.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>python313Packages.txtai</li>
    <li>python313Packages.txtai.dist</li>
    <li>rembg</li>
    <li>rembg.dist</li>
    <li>vectorcode</li>
    <li>vectorcode.dist</li>
    <li>vimPlugins.vectorcode-nvim</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 14 packages built:</summary>
  <ul>
    <li>python312Packages.django-mcp-server</li>
    <li>python312Packages.django-mcp-server.dist</li>
    <li>python312Packages.fastapi-mcp</li>
    <li>python312Packages.fastapi-mcp.dist</li>
    <li>python312Packages.fastmcp</li>
    <li>python312Packages.fastmcp.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.mcp</li>
    <li>python312Packages.mcp.dist</li>
    <li>python312Packages.mcpadapt</li>
    <li>python312Packages.mcpadapt.dist</li>
    <li>python312Packages.openai-agents</li>
    <li>python312Packages.openai-agents.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.smolagents</summary>
<pre>==================================== ERRORS ====================================
_____________________ ERROR collecting tests/test_types.py _____________________
ImportError while importing test module '/build/source/tests/test_types.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/2i76abyx06asxr2k3rgsp8zx647qj38h-python3-3.12.11/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_types.py:21: in <module>
    from transformers.testing_utils import (
E   ImportError: cannot import name 'require_soundfile' from 'transformers.testing_utils' (/nix/store/1qbmlivmw3zkgscmahsch9q6amnlvhz2-python3.12-transformers-4.54.0/lib/python3.12/site-packages/transformers/testing_utils.py)
=============================== warnings summary ===============================
tests/test_all_docs.py:107
  /build/source/tests/test_all_docs.py:107: PytestUnknownMarkWarning: Unknown pytest.mark.timeout - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.timeout(100)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR tests/test_types.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
============ 40 deselected, 1 warning, 1 error in 60.21s (0:01:00) =============</pre>
</details>
<details>
<summary>python313Packages.mcp</summary>
<pre>                        traceback.format_exception(
                            cm.unraisable.exc_type,
                            cm.unraisable.exc_value,
                            cm.unraisable.exc_traceback,
                        )
                    )
>                   warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
E                   pytest.PytestUnraisableExceptionWarning: Exception ignored in: <_io.FileIO [closed]>
E                   
E                   Traceback (most recent call last):
E                     File "/nix/store/wa8b4x0vsvxz1dvgwnm9gdwhvba7ka70-python3.13-pydantic-2.11.7/lib/python3.13/site-packages/pydantic/json_schema.py", line 319, in build_schema_type_to_method
E                       mapping[key] = getattr(self, method_name)
E                                      ~~~~~~~^^^^^^^^^^^^^^^^^^^
E                   ResourceWarning: unclosed file <_io.FileIO name=18 mode='rb' closefd=True>

/nix/store/irg9b8cr3rxqkmgvalddzh5ps6lxzva3-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/unraisableexception.py:85: PytestUnraisableExceptionWarning
=========================== short test summary info ============================
FAILED tests/client/test_stdio.py::TestChildProcessCleanup::test_early_parent_exit - AssertionError: Child should be writing
FAILED tests/issues/test_100_tool_listing.py::test_list_tools_returns_all_tools - pytest.PytestUnraisableExceptionWarning: Exception ignored in: <_io.FileIO ...
============= 2 failed, 514 passed, 2 skipped, 1 xfailed in 56.06s =============</pre>
</details>
</details>

bryanhonof commented:
Going to mark it as a draft until we fix more of the failing dependents.

bryanhonof commented:
P.S. if anyone wants to help contribute, let me know; I can add you as a maintainer to my fork for this branch.

bryanhonof committed:
python3Packages.gradio: 5.38.0 -> 5.38.2

bryanhonof committed:
oterm: allow newer versions of fastmcp dependency

We package a newer version of fastmcp already, but upstream seems to be limiting
it to 2.9.

GaetanLepage commented on pkgs/development/python-modules/fastmcp/default.nix:
```diff
@@ -64,16 +70,22 @@ buildPythonPackage rec {
     dirty-equals
     fastapi
     pytest-httpx
+    pytest-asyncio
     pytestCheckHook
     writableTmpDirAsHomeHook
+    email-validator
```
Same here

GaetanLepage replied:
Still not sorted alphabetically ;)

bryanhonof commented on pkgs/by-name/ot/oterm/package.nix:
```diff
@@ -19,6 +19,11 @@ python3Packages.buildPythonApplication rec {
+  postPatch = ''
+    substituteInPlace pyproject.toml --replace-fail "fastmcp>=2.8.0,<2.9" "fastmcp>=2.8.0"
+  '';
+
```
I have not, and I'm surprised I didn't see that list and didn't know about `pythonRelaxDeps `.
Thanks for pointing that out.

GaetanLepage commented on pkgs/development/python-modules/fastapi-mcp/default.nix:
```diff
@@ -64,6 +64,11 @@ buildPythonPackage rec {
   ];
 
+  disabledTestPaths = [
+    # Flakey, would try to allocate a port on Darwin
```
```suggestion
    # Flaky, would try to allocate a port on Darwin
```

GaetanLepage commented on pkgs/development/python-modules/fastmcp/default.nix:
```diff
@@ -64,16 +70,22 @@ buildPythonPackage rec {
 
-  disabledTestPaths = lib.optionals stdenv.hostPlatform.isDarwin [
+  disabledTestPaths = [
+    "tests/client/test_stdio.py"
+    "tests/utilities/test_mcp_config.py"
```
Can you add a comment stating why we are skipping these test paths please?

bryanhonof replied:
I'm also not sure if this is a Darwin-only problem. I don't have access to a Linux box currently to test on, sorry.

josh replied:
Just testing without the `disabledTestPaths`, I'm seeing similar fastmcp failures on both Mac and Linux. So the changes in a91ceabca6056d52520afd33fc0c436355e92bab make sense to me to make it build. Kinda interesting, I don't see anything after the "# RuntimeError: Server failed to start after 10 attempts" comment failing anymore on Darwin.

```
       > FAILED tests/client/test_stdio.py::TestKeepAlive::test_keep_alive_maintains_session_across_multiple_calls - RuntimeError: Client failed to connect: Connection closed
       > FAILED tests/client/test_stdio.py::TestKeepAlive::test_keep_alive_false_starts_new_session_across_multiple_calls - RuntimeError: Client failed to connect: Connection closed
       > FAILED tests/client/test_stdio.py::TestKeepAlive::test_keep_alive_starts_new_session_if_manually_closed - RuntimeError: Client failed to connect: Connection closed
       > FAILED tests/client/test_stdio.py::TestKeepAlive::test_keep_alive_maintains_session_if_reentered - RuntimeError: Client failed to connect: Connection closed
       > FAILED tests/client/test_stdio.py::TestKeepAlive::test_close_session_and_try_to_use_client_raises_error - RuntimeError: Client failed to connect: Connection closed
       > FAILED tests/utilities/test_mcp_config.py::test_multi_client - assert 0 == 2
       > FAILED tests/utilities/test_mcp_config.py::test_multi_client_with_logging - fastmcp.exceptions.ToolError: Unknown tool: test_server_log_test
       > FAILED tests/utilities/test_mcp_config.py::test_multi_client_with_elicitation - fastmcp.exceptions.ToolError: Unknown tool: test_server_elicit_test
```

GaetanLepage reviewed (commented):
Few minor comments. We are close ;)

bryanhonof committed:
mcp-proxy: disable tests that are failing

These started popping up when bumping the python3Packages.mcp version.

bryanhonof committed:
python3Packages.smolagents: disable failing tests

bryanhonof commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 428733`
Commit: `34f116df23261421562bff4e938082bfe0e232c2`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 55 packages built:</summary>
  <ul>
    <li>litellm</li>
    <li>litellm.dist</li>
    <li>mcp-proxy</li>
    <li>mcp-proxy.dist</li>
    <li>oterm</li>
    <li>oterm.dist</li>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python312Packages.django-mcp-server</li>
    <li>python312Packages.django-mcp-server.dist</li>
    <li>python312Packages.fastapi-mcp</li>
    <li>python312Packages.fastapi-mcp.dist</li>
    <li>python312Packages.fastmcp</li>
    <li>python312Packages.fastmcp.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.mcp</li>
    <li>python312Packages.mcp.dist</li>
    <li>python312Packages.mcpadapt</li>
    <li>python312Packages.mcpadapt.dist</li>
    <li>python312Packages.openai-agents</li>
    <li>python312Packages.openai-agents.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python313Packages.django-mcp-server</li>
    <li>python313Packages.django-mcp-server.dist</li>
    <li>python313Packages.fastapi-mcp</li>
    <li>python313Packages.fastapi-mcp.dist</li>
    <li>python313Packages.fastmcp</li>
    <li>python313Packages.fastmcp.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.mcp</li>
    <li>python313Packages.mcp.dist</li>
    <li>python313Packages.mcpadapt</li>
    <li>python313Packages.mcpadapt.dist</li>
    <li>python313Packages.openai-agents</li>
    <li>python313Packages.openai-agents.dist</li>
    <li>python313Packages.private-gpt</li>
    <li>python313Packages.private-gpt.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>vectorcode</li>
    <li>vectorcode.dist</li>
    <li>vimPlugins.vectorcode-nvim</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 55 packages built:</summary>
  <ul>
    <li>litellm</li>
    <li>litellm.dist</li>
    <li>mcp-proxy</li>
    <li>mcp-proxy.dist</li>
    <li>oterm</li>
    <li>oterm.dist</li>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python312Packages.django-mcp-server</li>
    <li>python312Packages.django-mcp-server.dist</li>
    <li>python312Packages.fastapi-mcp</li>
    <li>python312Packages.fastapi-mcp.dist</li>
    <li>python312Packages.fastmcp</li>
    <li>python312Packages.fastmcp.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.mcp</li>
    <li>python312Packages.mcp.dist</li>
    <li>python312Packages.mcpadapt</li>
    <li>python312Packages.mcpadapt.dist</li>
    <li>python312Packages.openai-agents</li>
    <li>python312Packages.openai-agents.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python313Packages.django-mcp-server</li>
    <li>python313Packages.django-mcp-server.dist</li>
    <li>python313Packages.fastapi-mcp</li>
    <li>python313Packages.fastapi-mcp.dist</li>
    <li>python313Packages.fastmcp</li>
    <li>python313Packages.fastmcp.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.mcp</li>
    <li>python313Packages.mcp.dist</li>
    <li>python313Packages.mcpadapt</li>
    <li>python313Packages.mcpadapt.dist</li>
    <li>python313Packages.openai-agents</li>
    <li>python313Packages.openai-agents.dist</li>
    <li>python313Packages.private-gpt</li>
    <li>python313Packages.private-gpt.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>vectorcode</li>
    <li>vectorcode.dist</li>
    <li>vimPlugins.vectorcode-nvim</li>
  </ul>
</details>

josh commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 428733`
Commit: `34f116df23261421562bff4e938082bfe0e232c2`

---
### `x86_64-linux`
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>home-assistant-component-tests.mcp</li>
    <li>home-assistant-component-tests.mcp_server</li>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
    <li>python313Packages.private-gpt</li>
    <li>python313Packages.private-gpt.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 55 packages built:</summary>
  <ul>
    <li>litellm</li>
    <li>litellm.dist</li>
    <li>mcp-proxy</li>
    <li>mcp-proxy.dist</li>
    <li>oterm</li>
    <li>oterm.dist</li>
    <li>python312Packages.django-mcp-server</li>
    <li>python312Packages.django-mcp-server.dist</li>
    <li>python312Packages.fastapi-mcp</li>
    <li>python312Packages.fastapi-mcp.dist</li>
    <li>python312Packages.fastmcp</li>
    <li>python312Packages.fastmcp.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.mcp</li>
    <li>python312Packages.mcp.dist</li>
    <li>python312Packages.mcpadapt</li>
    <li>python312Packages.mcpadapt.dist</li>
    <li>python312Packages.openai-agents</li>
    <li>python312Packages.openai-agents.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
    <li>python313Packages.django-mcp-server</li>
    <li>python313Packages.django-mcp-server.dist</li>
    <li>python313Packages.fastapi-mcp</li>
    <li>python313Packages.fastapi-mcp.dist</li>
    <li>python313Packages.fastmcp</li>
    <li>python313Packages.fastmcp.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.mcp</li>
    <li>python313Packages.mcp.dist</li>
    <li>python313Packages.mcpadapt</li>
    <li>python313Packages.mcpadapt.dist</li>
    <li>python313Packages.openai-agents</li>
    <li>python313Packages.openai-agents.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>python313Packages.txtai</li>
    <li>python313Packages.txtai.dist</li>
    <li>rembg</li>
    <li>rembg.dist</li>
    <li>vectorcode</li>
    <li>vectorcode.dist</li>
    <li>vimPlugins.vectorcode-nvim</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 24 packages failed to build:</summary>
  <ul>
    <li>home-assistant-component-tests.mcp</li>
    <li>home-assistant-component-tests.mcp_server</li>
    <li>oterm</li>
    <li>oterm.dist</li>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
    <li>python313Packages.fastmcp</li>
    <li>python313Packages.fastmcp.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.private-gpt</li>
    <li>python313Packages.private-gpt.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>python313Packages.txtai</li>
    <li>python313Packages.txtai.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 34 packages built:</summary>
  <ul>
    <li>litellm</li>
    <li>litellm.dist</li>
    <li>mcp-proxy</li>
    <li>mcp-proxy.dist</li>
    <li>python312Packages.django-mcp-server</li>
    <li>python312Packages.django-mcp-server.dist</li>
    <li>python312Packages.fastapi-mcp</li>
    <li>python312Packages.fastapi-mcp.dist</li>
    <li>python312Packages.fastmcp</li>
    <li>python312Packages.fastmcp.dist</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.mcp</li>
    <li>python312Packages.mcp.dist</li>
    <li>python312Packages.mcpadapt</li>
    <li>python312Packages.mcpadapt.dist</li>
    <li>python312Packages.openai-agents</li>
    <li>python312Packages.openai-agents.dist</li>
    <li>python313Packages.django-mcp-server</li>
    <li>python313Packages.django-mcp-server.dist</li>
    <li>python313Packages.fastapi-mcp</li>
    <li>python313Packages.fastapi-mcp.dist</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.mcp</li>
    <li>python313Packages.mcp.dist</li>
    <li>python313Packages.mcpadapt</li>
    <li>python313Packages.mcpadapt.dist</li>
    <li>python313Packages.openai-agents</li>
    <li>python313Packages.openai-agents.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>oterm</li>
    <li>oterm.dist</li>
    <li>python313Packages.fastmcp</li>
    <li>python313Packages.fastmcp.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 51 packages built:</summary>
  <ul>
    <li>litellm</li>
    <li>litellm.dist</li>
    <li>mcp-proxy</li>
    <li>mcp-proxy.dist</li>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python312Packages.django-mcp-server</li>
    <li>python312Packages.django-mcp-server.dist</li>
    <li>python312Packages.fastapi-mcp</li>
    <li>python312Packages.fastapi-mcp.dist</li>
    <li>python312Packages.fastmcp</li>
    <li>python312Packages.fastmcp.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.mcp</li>
    <li>python312Packages.mcp.dist</li>
    <li>python312Packages.mcpadapt</li>
    <li>python312Packages.mcpadapt.dist</li>
    <li>python312Packages.openai-agents</li>
    <li>python312Packages.openai-agents.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python313Packages.django-mcp-server</li>
    <li>python313Packages.django-mcp-server.dist</li>
    <li>python313Packages.fastapi-mcp</li>
    <li>python313Packages.fastapi-mcp.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.mcp</li>
    <li>python313Packages.mcp.dist</li>
    <li>python313Packages.mcpadapt</li>
    <li>python313Packages.mcpadapt.dist</li>
    <li>python313Packages.openai-agents</li>
    <li>python313Packages.openai-agents.dist</li>
    <li>python313Packages.private-gpt</li>
    <li>python313Packages.private-gpt.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>vectorcode</li>
    <li>vectorcode.dist</li>
    <li>vimPlugins.vectorcode-nvim</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>python312Packages.gradio</summary>
<pre>  File "/nix/store/9vpvnbxjb65bnakkkvb5r97gbbsbqlzk-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/main.py", line 975 in genitems
  File "/nix/store/9vpvnbxjb65bnakkkvb5r97gbbsbqlzk-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/main.py", line 809 in perform_collect
  File "/nix/store/9vpvnbxjb65bnakkkvb5r97gbbsbqlzk-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/main.py", line 347 in pytest_collection
  File "/nix/store/14s6c4wcr534964m1ldq5yrqhbx9q5gv-python3.12-pluggy-1.6.0/lib/python3.12/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/nix/store/14s6c4wcr534964m1ldq5yrqhbx9q5gv-python3.12-pluggy-1.6.0/lib/python3.12/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/nix/store/14s6c4wcr534964m1ldq5yrqhbx9q5gv-python3.12-pluggy-1.6.0/lib/python3.12/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/nix/store/9vpvnbxjb65bnakkkvb5r97gbbsbqlzk-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/main.py", line 336 in _main
  File "/nix/store/9vpvnbxjb65bnakkkvb5r97gbbsbqlzk-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/main.py", line 283 in wrap_session
  File "/nix/store/9vpvnbxjb65bnakkkvb5r97gbbsbqlzk-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/main.py", line 330 in pytest_cmdline_main
  File "/nix/store/14s6c4wcr534964m1ldq5yrqhbx9q5gv-python3.12-pluggy-1.6.0/lib/python3.12/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/nix/store/14s6c4wcr534964m1ldq5yrqhbx9q5gv-python3.12-pluggy-1.6.0/lib/python3.12/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/nix/store/14s6c4wcr534964m1ldq5yrqhbx9q5gv-python3.12-pluggy-1.6.0/lib/python3.12/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/nix/store/9vpvnbxjb65bnakkkvb5r97gbbsbqlzk-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/config/__init__.py", line 175 in main
  File "/nix/store/9vpvnbxjb65bnakkkvb5r97gbbsbqlzk-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/config/__init__.py", line 201 in console_main
  File "/nix/store/9vpvnbxjb65bnakkkvb5r97gbbsbqlzk-python3.12-pytest-8.3.5/lib/python3.12/site-packages/pytest/__main__.py", line 9 in <module>
  File "<frozen runpy>", line 88 in _run_code
  File "<frozen runpy>", line 198 in _run_module_as_main

Extension modules: _brotli, charset_normalizer.md, requests.packages.charset_normalizer.md, requests.packages.chardet.md, yaml._yaml, websockets.speedups, numpy._core._multiarray_umath, numpy.linalg._umath_linalg, PIL._imaging, markupsafe._speedups, PIL._avif, PIL._imagingmath, PIL._webp, psutil._psutil_linux, psutil._psutil_posix, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._pcg64, numpy.random._mt19937, numpy.random._generator, numpy.random._philox, numpy.random._sfc64, numpy.random.mtrand, pandas._libs.tslibs.ccalendar, pandas._libs.tslibs.np_datetime, pandas._libs.tslibs.dtypes, pandas._libs.tslibs.base, pandas._libs.tslibs.nattype, pandas._libs.tslibs.timezones, pandas._libs.tslibs.fields, pandas._libs.tslibs.timedeltas, pandas._libs.tslibs.tzconversion, pandas._libs.tslibs.timestamps, pandas._libs.properties, pandas._libs.tslibs.offsets, pandas._libs.tslibs.strptime, pandas._libs.tslibs.parsing, pandas._libs.tslibs.conversion, pandas._libs.tslibs.period, pandas._libs.tslibs.vectorized, pandas._libs.ops_dispatch, pandas._libs.missing, pandas._libs.hashtable, pandas._libs.algos, pandas._libs.interval, pandas._libs.lib, pandas._libs.ops, pandas._libs.hashing, pandas._libs.arrays, pandas._libs.tslib, pandas._libs.sparse, pandas._libs.internals, pandas._libs.indexing, pandas._libs.index, pandas._libs.writers, pandas._libs.join, pandas._libs.window.aggregations, pandas._libs.window.indexers, pandas._libs.reshape, pandas._libs.groupby, pandas._libs.json, pandas._libs.parsers, pandas._libs.testing (total: 64)
/nix/store/4ga83x05amks5a16fs3wjpdy5c1gfxya-pytest-check-hook/nix-support/setup-hook: line 23: 12910 Aborted                    (core dumped) /nix/store/4k1n9qx7di8k7p2fq14phwhw9bzdfzki-python3-3.12.11/bin/python3.12 "${flagsArray[@]}"</pre>
</details>
<details>
<summary>python313Packages.fastmcp</summary>
<pre>
/nix/store/3dqjlnj2iaiaglkbg8qph7jd219n0sfh-python3.13-fastmcp-2.10.6/lib/python3.13/site-packages/fastmcp/client/client.py:406: RuntimeError
----------------------------- Captured stdout call -----------------------------
INFO:     127.0.0.1:55578 - "GET /sse/ HTTP/1.1" 200 OK
INFO:     127.0.0.1:55592 - "POST /messages/?session_id=b42007195e8e49088290194c3df5c222 HTTP/1.1" 202 Accepted
=============================== warnings summary ===============================
../../nix/store/kmyxbspnal8zpwvq2l2njhrv7l5ypk4b-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/config/__init__.py:1441
  /nix/store/kmyxbspnal8zpwvq2l2njhrv7l5ypk4b-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/config/__init__.py:1441: PytestConfigWarning: Unknown config option: env
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

../../nix/store/kmyxbspnal8zpwvq2l2njhrv7l5ypk4b-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/config/__init__.py:1441
  /nix/store/kmyxbspnal8zpwvq2l2njhrv7l5ypk4b-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/config/__init__.py:1441: PytestConfigWarning: Unknown config option: timeout
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/client/test_sse.py::TestTimeout::test_timeout - RuntimeError: Client failed to connect: Timed out while waiting for respons...
===== 1 failed, 2026 passed, 9 deselected, 8 xfailed, 2 warnings in 28.11s =====</pre>
</details>
<details>
<summary>python313Packages.gradio</summary>
<pre>  File "/nix/store/kmyxbspnal8zpwvq2l2njhrv7l5ypk4b-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/main.py", line 975 in genitems
  File "/nix/store/kmyxbspnal8zpwvq2l2njhrv7l5ypk4b-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/main.py", line 809 in perform_collect
  File "/nix/store/kmyxbspnal8zpwvq2l2njhrv7l5ypk4b-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/main.py", line 347 in pytest_collection
  File "/nix/store/63xwihji3yxhl7mnail9kqv12hhnrcip-python3.13-pluggy-1.6.0/lib/python3.13/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/nix/store/63xwihji3yxhl7mnail9kqv12hhnrcip-python3.13-pluggy-1.6.0/lib/python3.13/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/nix/store/63xwihji3yxhl7mnail9kqv12hhnrcip-python3.13-pluggy-1.6.0/lib/python3.13/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/nix/store/kmyxbspnal8zpwvq2l2njhrv7l5ypk4b-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/main.py", line 336 in _main
  File "/nix/store/kmyxbspnal8zpwvq2l2njhrv7l5ypk4b-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/main.py", line 283 in wrap_session
  File "/nix/store/kmyxbspnal8zpwvq2l2njhrv7l5ypk4b-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/main.py", line 330 in pytest_cmdline_main
  File "/nix/store/63xwihji3yxhl7mnail9kqv12hhnrcip-python3.13-pluggy-1.6.0/lib/python3.13/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/nix/store/63xwihji3yxhl7mnail9kqv12hhnrcip-python3.13-pluggy-1.6.0/lib/python3.13/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/nix/store/63xwihji3yxhl7mnail9kqv12hhnrcip-python3.13-pluggy-1.6.0/lib/python3.13/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/nix/store/kmyxbspnal8zpwvq2l2njhrv7l5ypk4b-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/config/__init__.py", line 175 in main
  File "/nix/store/kmyxbspnal8zpwvq2l2njhrv7l5ypk4b-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/config/__init__.py", line 201 in console_main
  File "/nix/store/kmyxbspnal8zpwvq2l2njhrv7l5ypk4b-python3.13-pytest-8.3.5/lib/python3.13/site-packages/pytest/__main__.py", line 9 in <module>
  File "<frozen runpy>", line 88 in _run_code
  File "<frozen runpy>", line 198 in _run_module_as_main

Extension modules: _brotli, charset_normalizer.md, requests.packages.charset_normalizer.md, requests.packages.chardet.md, yaml._yaml, websockets.speedups, numpy._core._multiarray_umath, numpy.linalg._umath_linalg, PIL._imaging, audioop._audioop, markupsafe._speedups, PIL._avif, PIL._imagingmath, PIL._webp, psutil._psutil_linux, psutil._psutil_posix, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._pcg64, numpy.random._mt19937, numpy.random._generator, numpy.random._philox, numpy.random._sfc64, numpy.random.mtrand, pandas._libs.tslibs.ccalendar, pandas._libs.tslibs.np_datetime, pandas._libs.tslibs.dtypes, pandas._libs.tslibs.base, pandas._libs.tslibs.nattype, pandas._libs.tslibs.timezones, pandas._libs.tslibs.fields, pandas._libs.tslibs.timedeltas, pandas._libs.tslibs.tzconversion, pandas._libs.tslibs.timestamps, pandas._libs.properties, pandas._libs.tslibs.offsets, pandas._libs.tslibs.strptime, pandas._libs.tslibs.parsing, pandas._libs.tslibs.conversion, pandas._libs.tslibs.period, pandas._libs.tslibs.vectorized, pandas._libs.ops_dispatch, pandas._libs.missing, pandas._libs.hashtable, pandas._libs.algos, pandas._libs.interval, pandas._libs.lib, pandas._libs.ops, pandas._libs.hashing, pandas._libs.arrays, pandas._libs.tslib, pandas._libs.sparse, pandas._libs.internals, pandas._libs.indexing, pandas._libs.index, pandas._libs.writers, pandas._libs.join, pandas._libs.window.aggregations, pandas._libs.window.indexers, pandas._libs.reshape, pandas._libs.groupby, pandas._libs.json, pandas._libs.parsers, pandas._libs.testing (total: 65)
/nix/store/qn5497xhhrdpb6nj14c3r55g7vamp9hw-pytest-check-hook/nix-support/setup-hook: line 23: 12910 Aborted                    (core dumped) /nix/store/cr7c6cf8y7sbqwq9bj7lrm977x7c4mv9-python3-3.13.5/bin/python3.13 "${flagsArray[@]}"</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python313Packages.fastmcp</summary>
<pre>                        f"Client failed to connect: {exception}"
                    ) from exception
E                   RuntimeError: Client failed to connect: Timed out while waiting for response to ClientRequest. Waited 0.01 seconds.

/nix/store/16x9m5ayf37i9p00p9hpknfs7ljyb6fb-python3.13-fastmcp-2.10.6/lib/python3.13/site-packages/fastmcp/client/client.py:406: RuntimeError
=============================== warnings summary ===============================
../../../../nix/store/s8fyrhxln9cds1fckd37vazh5xfpknhi-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/config/__init__.py:1441
  /nix/store/s8fyrhxln9cds1fckd37vazh5xfpknhi-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/config/__init__.py:1441: PytestConfigWarning: Unknown config option: env
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

../../../../nix/store/s8fyrhxln9cds1fckd37vazh5xfpknhi-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/config/__init__.py:1441
  /nix/store/s8fyrhxln9cds1fckd37vazh5xfpknhi-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/config/__init__.py:1441: PytestConfigWarning: Unknown config option: timeout
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/client/test_client.py::TestTimeout::test_timeout_tool_call_overrides_client_timeout_even_if_lower - RuntimeError: Client failed to connect: Timed out while waiting for respons...
= 1 failed, 1944 passed, 9 deselected, 7 xfailed, 2 warnings in 93.52s (0:01:33) =</pre>
</details>
</details>

bryanhonof commented:
Not quite sure what's up with the `gradio` package... Could use some help debugging that one.

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 428733`
Commit: `ffec6a93183648193618d1d78f54e7425d8caf68`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>home-assistant-component-tests.mcp</li>
    <li>home-assistant-component-tests.mcp_server</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 61 packages built:</summary>
  <ul>
    <li>litellm</li>
    <li>litellm.dist</li>
    <li>mcp-proxy</li>
    <li>mcp-proxy.dist</li>
    <li>oterm</li>
    <li>oterm.dist</li>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python312Packages.django-mcp-server</li>
    <li>python312Packages.django-mcp-server.dist</li>
    <li>python312Packages.fastapi-mcp</li>
    <li>python312Packages.fastapi-mcp.dist</li>
    <li>python312Packages.fastmcp</li>
    <li>python312Packages.fastmcp.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.mcp</li>
    <li>python312Packages.mcp.dist</li>
    <li>python312Packages.mcpadapt</li>
    <li>python312Packages.mcpadapt.dist</li>
    <li>python312Packages.openai-agents</li>
    <li>python312Packages.openai-agents.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
    <li>python313Packages.django-mcp-server</li>
    <li>python313Packages.django-mcp-server.dist</li>
    <li>python313Packages.fastapi-mcp</li>
    <li>python313Packages.fastapi-mcp.dist</li>
    <li>python313Packages.fastmcp</li>
    <li>python313Packages.fastmcp.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.mcp</li>
    <li>python313Packages.mcp.dist</li>
    <li>python313Packages.mcpadapt</li>
    <li>python313Packages.mcpadapt.dist</li>
    <li>python313Packages.openai-agents</li>
    <li>python313Packages.openai-agents.dist</li>
    <li>python313Packages.private-gpt</li>
    <li>python313Packages.private-gpt.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>python313Packages.txtai</li>
    <li>python313Packages.txtai.dist</li>
    <li>rembg</li>
    <li>rembg.dist</li>
    <li>vectorcode</li>
    <li>vectorcode.dist</li>
    <li>vimPlugins.vectorcode-nvim</li>
  </ul>
</details>

GaetanLepage commented:
CI is failing :/
Maybe try to rebase on `master` and force-push.

GaetanLepage commented:
Nevermind, failure is caused by rate limitation, this has nothing to do with your changes:
```
Error: Unhandled error: HttpError: API rate limit exceeded for installation. If you reach out to GitHub Support for help, please include the request ID 1C01:1F61A:B10827:1663625:688AAA77 and timestamp 2025-07-30 23:27:51 UTC.
```

bryanhonof commented:
Ah, I found why you were getting that one error.
It's coming from the home-assistant component test: https://github.com/NixOS/nixpkgs/blob/e5b6260eb9c628201ba3341a0d6d2b7af2fee2e4/pkgs/servers/home-assistant/default.nix#L143-L150

bryanhonof committed:
home-assistant: ignore Pydantic deprecation warnigns for mcp package

We bumped the python3Packages.mcp package, and were able to remove this set of
flags due to the Pydantic library also being updated. However, since
home-assistant still seems to be using 1.5.0, and overriding it here, these
flags are still needed in this particular case.

mweinelt commented:
> However, since home-manager still seems to be using 1.5.0, and overriding it here, these flags are still needed in this particular case.

Home-manager vs Home Assistant.

bryanhonof commented:
> > However, since home-manager still seems to be using 1.5.0, and overriding it here, these flags are still needed in this particular case.
> 
> Home-manager vs Home Assistant.

Whups. 😅

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 428733`
Commit: `9656b9a94a0b98ca612467faf82c4f9c8a010c4b`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>home-assistant-component-tests.mcp</li>
    <li>home-assistant-component-tests.mcp_server</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 55 packages built:</summary>
  <ul>
    <li>litellm</li>
    <li>litellm.dist</li>
    <li>mcp-proxy</li>
    <li>mcp-proxy.dist</li>
    <li>oterm</li>
    <li>oterm.dist</li>
    <li>python312Packages.django-mcp-server</li>
    <li>python312Packages.django-mcp-server.dist</li>
    <li>python312Packages.fastapi-mcp</li>
    <li>python312Packages.fastapi-mcp.dist</li>
    <li>python312Packages.fastmcp</li>
    <li>python312Packages.fastmcp.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.mcp</li>
    <li>python312Packages.mcp.dist</li>
    <li>python312Packages.mcpadapt</li>
    <li>python312Packages.mcpadapt.dist</li>
    <li>python312Packages.openai-agents</li>
    <li>python312Packages.openai-agents.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
    <li>python313Packages.django-mcp-server</li>
    <li>python313Packages.django-mcp-server.dist</li>
    <li>python313Packages.fastapi-mcp</li>
    <li>python313Packages.fastapi-mcp.dist</li>
    <li>python313Packages.fastmcp</li>
    <li>python313Packages.fastmcp.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.mcp</li>
    <li>python313Packages.mcp.dist</li>
    <li>python313Packages.mcpadapt</li>
    <li>python313Packages.mcpadapt.dist</li>
    <li>python313Packages.openai-agents</li>
    <li>python313Packages.openai-agents.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>python313Packages.txtai</li>
    <li>python313Packages.txtai.dist</li>
    <li>rembg</li>
    <li>rembg.dist</li>
    <li>vectorcode</li>
    <li>vectorcode.dist</li>
    <li>vimPlugins.vectorcode-nvim</li>
  </ul>
</details>

GaetanLepage commented:
> I'm not sure if it's just me who's either doing something wrong or losing my mind, but it seems like `python3Packages.mcp` just keeps popping up with new flaky tests.

Yes, that was it!

GaetanLepage reviewed (approved):
LGTM! Thanks!

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 428733`
Commit: `177392b7210f935a699ae6f8d2ad10426cbc4f35`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 57 packages built:</summary>
  <ul>
    <li>home-assistant-component-tests.mcp</li>
    <li>home-assistant-component-tests.mcp_server</li>
    <li>litellm</li>
    <li>litellm.dist</li>
    <li>mcp-proxy</li>
    <li>mcp-proxy.dist</li>
    <li>oterm</li>
    <li>oterm.dist</li>
    <li>python312Packages.django-mcp-server</li>
    <li>python312Packages.django-mcp-server.dist</li>
    <li>python312Packages.fastapi-mcp</li>
    <li>python312Packages.fastapi-mcp.dist</li>
    <li>python312Packages.fastmcp</li>
    <li>python312Packages.fastmcp.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.mcp</li>
    <li>python312Packages.mcp.dist</li>
    <li>python312Packages.mcpadapt</li>
    <li>python312Packages.mcpadapt.dist</li>
    <li>python312Packages.openai-agents</li>
    <li>python312Packages.openai-agents.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
    <li>python313Packages.django-mcp-server</li>
    <li>python313Packages.django-mcp-server.dist</li>
    <li>python313Packages.fastapi-mcp</li>
    <li>python313Packages.fastapi-mcp.dist</li>
    <li>python313Packages.fastmcp</li>
    <li>python313Packages.fastmcp.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.mcp</li>
    <li>python313Packages.mcp.dist</li>
    <li>python313Packages.mcpadapt</li>
    <li>python313Packages.mcpadapt.dist</li>
    <li>python313Packages.openai-agents</li>
    <li>python313Packages.openai-agents.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>python313Packages.txtai</li>
    <li>python313Packages.txtai.dist</li>
    <li>rembg</li>
    <li>rembg.dist</li>
    <li>vectorcode</li>
    <li>vectorcode.dist</li>
    <li>vimPlugins.vectorcode-nvim</li>
  </ul>
</details>

NickCao commented on pkgs/development/python-modules/fastmcp/default.nix:
```diff
@@ -49,10 +53,12 @@ buildPythonPackage rec {
+    pyperclip
     python-dotenv
     rich
     typer
```
No longer depends on `typer`, also new dependency on `pydantic` and `pydantic.optional-dependencies.email`

NickCao reviewed (commented):
The amount of new failing tests is concerning, but otherwise LGTM.

josh commented:
💥 Nice work @bryanhonof!
